evm_loader/lib/src/account_storage.rs

```diff
@@ -973,17 +967,25 @@ impl<T: Rpc> AccountStorage for EmulatorAccountStorage<'_, T> {
     async fn nonce(&self, address: Address, chain_id: u64) -> u64 {
         info!("nonce {address}  {chain_id}");
 
-        self.ethereum_balance_map_or(address, chain_id, u64::default(), |account| account.nonce())
-            .await
-            .unwrap()
+        self.ethereum_balance_map_or(
+            address,
+            chain_id,
+            u64::default(),
+            |account: &BalanceAccount| account.nonce(),
```

Comment:
Why not just `BalanceAccount::nonce`, as this is also what `cargo clippy` suggests? Same comment for the other similar usages too.

Reply:
it throws errors in this case about mismatching types and requires some refactoring which I would like to avoid in this PR